### PR TITLE
feature: some examples how to include resilience

### PR DESCRIPTION
**Pull request by FrankEssenberger**
Added two examples for resilience: retry and circuit breaker. In order to execute it locally you must include a `.npmrc` with the internal nexus or wait until the 10th of feb until the 2.0 business partner VDM is availible.

### CODE REVIEW

File: samples/resilience-examples/src/test-util.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  }
+};
+
+export function mockGetAllRequest(times: number): void {
+  nock(destinationUrl).get(/.*/).times(times).reply(200, sampleResponse);
+}
+
+export async function getAllBusinessPartner(
+  top: number
+): Promise<BusinessPartner[]> {
```
Comment (florian-richter) [inline samples/resilience-examples/src/test-util.ts:27]: It's a bit confusing to me that this is part of `test-util.ts`. Also I would like to rename the parameter to something that is clearer without knowing the implementation (e.g. `filterTop` or maybe `filterTopBuPa`).

**Review by florian-richter — COMMENTED** [review]
Some minor comments, overall pretty clear and (in my opinion) helpful 😊

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -23,15 +23,25 @@ jobs:
...
          cd samples/cds-sample-application
           npm install
           npm run ci-build
-      - name: e2e Tests
+      - name: CDS samples - e2e Tests
         run:  |
          cd samples/cds-sample-application
          npm run test:e2e
+      - name: Resilience examples - Install dependencies and build package
+          run:  |
```
Comment (deekshas8) [inline .github/workflows/build.yml:36]: There is a syntax error(indentation) in line 36 so the workflow run is failing

File: samples/resilience-examples/src/circuit-breaker.spec.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+  afterEach(() => {
+    nock.cleanAll();
+  });
+
+  afterAll(() => {
+    unmockAllTestDestinations();
+  });
+
+  it('does not open the cuircuit breaker for no failed requests.', async () => {
```
Comment (deekshas8) [inline samples/resilience-examples/src/circuit-breaker.spec.ts:28]:
```suggestion
  it('does not open the circuit breaker for no failed requests.', async () => {
```

File: samples/resilience-examples/src/circuit-breaker.spec.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    const spy = jest.spyOn(logger, 'warn');
+    await Promise.all(
+      numberWorkingRequests.map(() =>
+        getAllBusinessPartnerWithCircuitBreaker(1)
+      )
+    );
+    expect(spy).toHaveBeenCalledTimes(0);
+  });
+
+  it('opens the cuircuit after a few failed requests', async () => {
```
Comment (deekshas8) [inline samples/resilience-examples/src/circuit-breaker.spec.ts:38]:
```suggestion
  it('opens the circuit after a few failed requests', async () => {
```

File: samples/resilience-examples/src/test-util.ts (hunk 1 above)
Comment (FrankEssenberger) [reply samples/resilience-examples/src/test-util.ts:27]: hm, about the location of the method I agree. Perhaps just copy past it twice?

**Review by deekshas8 — APPROVED** [review]
LGTM

**Pull request merged**